### PR TITLE
Use JDK 17 for Benchmark Images

### LINKED ISSUE #8091: Upgrade Benchmark Images to JDK 17
**Description**

The starter and worker images we use in benchmarks can be upgraded to JDK 17.

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

This upgrades the images used for the starter and worker to use JDK 17. We also build the starter and worker inside of a docker container which is now also using JDK 17.

## Related issues

closes #8091

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Review by npepinpe — APPROVED** [review]
For the worker/starter, since we don't care about backwards compatibility, maybe we can just aim for the smallest image? In this case, something like `azul/zulu-openjdk-alpine:17-jre-headless`? Just a thought. We also were thinking of using Jib for these, but I guess that increase the scope quite a bit, so no worries. We could also already bump the language level of the benchmark project to 17?

**Comment by oleschoenburg** [comment]
> We could also already bump the language level of the benchmark project to 17

I'm not sure how that'll work with projects using zeebe as a dependency. Of course no one (hopefully 😅) directly relies on the benchmark module and we don't export it in the bom. If you don't think this is a problem I'll gladly switch to language level 17

> azul/zulu-openjdk-alpine:17-jre-headless

I'll try that out!

**Comment by npepinpe** [comment]
No projects depend on the benchmark, they're supposed to be standalone applications used only by us.

**Comment by oleschoenburg** [comment]
@npepinpe I've upgraded the benchmark module to Java 17 😎 and tested the worker and starter manually.

File: benchmarks/project/pom.xml [hunk 1]
```diff
@@ -128,6 +129,17 @@
         <artifactId>fmt-maven-plugin</artifactId>
       </plugin>
 
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
```
Comment (npepinpe) [inline benchmarks/project/pom.xml:134]: This plugin doesn't work Java 17? Is this something that was already on our radar? I'm not sure we'll want to keep it disabled, no?

**Review by npepinpe — APPROVED** [review]
Looks good, just had one question, but I don't think it's a blocker for the benchmark app

File: benchmarks/project/pom.xml (hunk 1 above)
Comment (oleschoenburg) [reply benchmarks/project/pom.xml:134]: It was already on my radar. Maybe it was badly worded by me, what doesn't work is `maven-dependency-analyzer`. We already talked about this and came to the conclusion that it's okay to disable this for the modules that we upgrade to Java 17 und re-enable when a fixed version is released.

**Comment by oleschoenburg** [comment]
bors merge

**Comment by oleschoenburg** [comment]
I just noticed that LGTM does not run through. I'll see if that's something we can fix here

**Comment by oleschoenburg** [comment]
@npepinpe LGTM does not support Java 17: https://lgtm.com/help/lgtm/java-extraction
Do you see this as a blocker?

**Comment by npepinpe** [comment]
It doesn't support it out of the box, or it's plain impossible for it to analyze any Java 17 project?

**Comment by npepinpe** [comment]
I'm trying to figure out the difference between LGTM and the CodeQL workflows that GitHub now provides, actually, since LGTM was acquired by GitHub and then their CodeQL stuff made available :thinking:

**Comment by npepinpe** [comment]
Let's discuss it tomorrow. For now, you can lower the language level for these modules, and we'll figure it out after.

**Comment by oleschoenburg** [comment]
bors merge

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**